meta-facebook/yv35-gl/src/platform/plat_class.c

```diff
@@ -182,7 +182,36 @@ void init_platform_config()
 	if (status)
 		LOG_ERR("failed to get class type from cpld, ret %d", status);
 	class_type = msg.data[0];
-	_1ou_status.present = ((class_type & BIT(2)) ? false : true);
+
+	_1ou_status.present = false;
+
+	if (!(class_type & BIT(2))) {
+		if ((class_type & BIT(0)) && (class_type & BIT(1)) && (class_type & BIT(4))) {
+			bool e1s_card_change_flag = false;
```

Comment:
Do we really need "e1s_card_change_flag", can't we have a check on 
```
_1ou_card_mapping_table[i].card_type != TYPE_1OU_EXP_WITH_E1S
```
when printing error at line# 203?

Reply:
We remove the flag and check if the for loop is successful by "_1ou_card_mapping_table[count].card_type != TYPE_1OU_EXP_WITH_E1S"

Could you please confirm? 
Thanks.